### PR TITLE
Fix #124486: Add "French Horn" instrument, fix #12489: General "Percussion" staff

### PR DESCRIPTION
**Pull request by IsaacWeiss**
"French Horn" same as existing "Horn in F"; "Percussion" five line staff, combining orchestral unpitched percussion instruments

Also incidentally rename “Marching Percussion” category to more specific “Battery Percussion,” and correct “High Hat” to “Hi-Hat” in a couple places.

### CODE REVIEW

**Comment by IsaacWeiss** [comment]
@Jojo-Schmitz, thanks for reminding me what I forget every time (though your comment seems to have disappeared). The script also made a change in the guitar area; could you take a look at that?

**Comment by Jojo-Schmitz** [comment]
Not sure, at least Travesty isn't happy about it?

**Comment by Jojo-Schmitz** [comment]
Travis...  Damn aurocorrection

File: share/instruments/instruments.xml [hunk 1]
```diff
@@ -2823,6 +2823,23 @@
                         <program value="60"/>
                   </Channel>
             </Instrument>
+            <Instrument id="horn">
```
Comment (lasconic) [inline share/instruments/instruments.xml]: We already have a French Horn. It called Horn in F and already use the id "horn" and the musicxmlID brass.french-horn. If those are the same instrument "Horn in F" and "french horn" then let's pick one and not add another entry.

**Comment by IsaacWeiss** [comment]
Why? We have both contrabass and double bass, for good reason. I feel this is the same.

**Comment by lasconic** [comment]
If it's the same instrument, order will matter and you can use `init`.

**Comment by mgavioli** [comment]
IIRC, a difference was intended between contrabass and double bass, one being the 'classic' use of it defaulting to arco and the other the jazz usage defaulting to pizzicato (I never remember which is which).

Anyway, adding another name for the same instrument would possibly raise problems with translations, as most languages have only one name for the instrument and having two would result in duplicate entries with the same name (for instance, Italian only has the name "corno", so there would be two "corno" entries with the same name and the same contents).

**Comment by lasconic** [comment]
@mgavioli Good to see you and your wisdom around again!
Actually we have *contrabass* and *double bass* which are exactly the same and we have *acoustic bass* for the "plucked" version.

**Comment by MarcSabatella** [comment]
Actually, it's "acoustic bass" that is the pizz-default jazz version; contrabass and double bass are just alternate names for the same instrument.  Both contrabass and double bass exist in our list because both names are standard, correct, and in common use in published music.  Whereas slang terms like "doghouse" or "big a$$ violin" are not, so we don't include them as additional aliases, any more than we include "licorice stick" or "five easy pieces" as aliases for clarinet.  "French horn" is essentially in this same category, but it's less *obviously* just a slang term - to the point where many people don't actually know the more proper name.  In that sense it is maybe more like "fiddle"., although that's actually "official" in *some* circles at least, whereas "French horn" really isn't.  But if including the alias helps people who are not aware of the proper name to find the instrument, I have no real problem with it.  I kind of wish there was a way to link it, though - list is as "French horn (see Horn in F)" or something like that.

**Comment by MarcSabatella** [comment]
(Sorry, posts crossed)

**Comment by IsaacWeiss** [comment]
@lasconic, I think I see what you mean about init. I'll change it so it works the same way, and hopefully that will also clear up those curious test failures.

**Comment by IsaacWeiss** [comment]
So why are the file import tests picking the "French" instead of the "in F"?

**Comment by Jojo-Schmitz** [comment]
Alphabetical order?

**Comment by Jojo-Schmitz** [comment]
BTW the translation problem exists for German and the contrabass / double bass. Both translate to Kontrabass

**Comment by mgavioli** [comment]
@Jojo-Schmitz : "_BTW the translation problem exists for German and the contrabass / double bass. Both translate to Kontrabass_": and in Italian too (at least I assume it exists, as I never use the Italian version, but as far as I know the instrument has no other Italian name than "contrabbasso").

**Comment by lasconic** [comment]
Same for french I believe. We could find and hide duplicate entries if necessary...

**Comment by IsaacWeiss** [comment]
In the default English version, I am slightly bothered that the addition of "French Horn" makes it preferred over "Horn in F" for MIDI import. Is there anything I can do to change that?

**Comment by lasconic** [comment]
My understanding is that we are taking the first instrument which match the MIDI program in the MIDI file. 
Interestingly enough, I can't reproduce the failure when running the test locally but I have another failure with Trumpet/cornet in `instrument_channels.mscx`.... Maybe we don't really keep the order of the instruments.xml file...

**Comment by IsaacWeiss** [comment]
If there's no way to "prefer" Horn in F, maybe this is a bad idea.

**Comment by mgavioli** [comment]
I believe a way to go could be extending the intruments.xml structure to add the concept of **synonyms** (possibly with a preferred one), rather than adding a number of spurious entries to accommodate the peculiarities of a single language.

So far we have only dealt with English synonyms, but what shall we do once similar cases should be found in French or German or Italian or Chinese? Should the translator disregard the specifics of his language or add duplicated entries in all other languages?

**Comment by IsaacWeiss** [comment]
See #2829 for a version without the French Horn.

**Comment by IsaacWeiss** [comment]
#2829 being merged, closing this.

**Pull request closed without merging**